### ISSUE TITLE
missing `network_identifier` check

### ISSUE THREAD

**Issue by MACS-J1149** [monadicus/mentat#44]
a check is never performed to make sure the `network_identifier` field is correct before making a request to a node, meaning you can specify `testnet` in your request to mentat but mentat will blindly pass on the request to whatever node is running without checking if its actually `testnet` first. this would lead to either incorrect data being returned or an error being returned from a seemingly valid request.

```
"network_identifier": {
    "blockchain": "bitcoin",
    "network": "mainnet",
    "sub_network_identifier": {
        "network": "shard 1",
        "metadata": {
            "producer": "0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5"
        }
    }
},
```

so basically, we need to add checks for each endpoint in mentat to make sure the blockchain/network being requested actually exists and return an error if it doesn't. if i recall correctly this info should be available as environment variables, so this would be as simple as adding a method on the `NetworkIdentifier` struct that performs this check then calling that method whenever `network_identifier` is passed

**Comment by mowkoshka** [comment]
We can do this easily with just a middleware :)

**Comment by MACS-J1149** [comment]
@mowkoshka my idea was to just add the checks in `mentat\src\api` similar to how we handle online and offline currently. how are you suggesting we do it?

**Comment by mowkoshka** [comment]
Ah, yea we can do it in the caller section as well! Either works.

I just thought a middleware might be more generic i.e. check if the payload has a field named network_identifier if so validate. If valid continue on to request otherwise error out.
